File: sources/LiquidSense Release/me/aquavit/liquidsense/module/modules/player/Regen.java
package me.aquavit.liquidsense.module.modules.player;

import me.aquavit.liquidsense.utils.entity.MovementUtils;
import me.aquavit.liquidsense.event.EventTarget;
import me.aquavit.liquidsense.event.events.UpdateEvent;
import me.aquavit.liquidsense.module.Module;
import me.aquavit.liquidsense.module.ModuleCategory;
import me.aquavit.liquidsense.module.ModuleInfo;
import me.aquavit.liquidsense.value.BoolValue;
import me.aquavit.liquidsense.value.IntegerValue;
import me.aquavit.liquidsense.value.ListValue;
import net.minecraft.network.play.client.C03PacketPlayer;

@ModuleInfo(name = "Regen", description = "Regenerates your health much faster.", category = ModuleCategory.PLAYER)
public class Regen extends Module {

    private ListValue modeValue = new ListValue("Mode", new String[]{"Vanilla", "Spartan"}, "Vanilla");
    private IntegerValue healthValue = new IntegerValue("Health", 18, 0, 20);
    private IntegerValue foodValue = new IntegerValue("Food", 18, 0, 20);
    private IntegerValue speedValue = new IntegerValue("Speed", 100, 1, 100);
    private BoolValue noAirValue = new BoolValue("NoAir", false);

    private boolean resetTimer;


    @EventTarget
    public void onUpdate(UpdateEvent event){
        if (resetTimer) mc.timer.timerSpeed = 1F;

        if ((!noAirValue.get() || mc.thePlayer.onGround) && !mc.thePlayer.capabilities.isCreativeMode &&
                mc.thePlayer.getFoodStats().getFoodLevel() > foodValue.get() && mc.thePlayer.isEntityAlive() && mc.thePlayer.getHealth() < healthValue.get()) {
            switch (modeValue.get()){
                case "Vanilla": {
                    for (int i=0; i<=speedValue.get();i++){
                        mc.getNetHandler().addToSendQueue(new C03PacketPlayer());
                    }
                    break;
                }
                case "Spartan": {
                    if (MovementUtils.isMoving() || !mc.thePlayer.onGround) return;

                    for (int i=0; i<=9;i++){
                        mc.getNetHandler().addToSendQueue(new C03PacketPlayer());
                    }

                    mc.timer.timerSpeed = 0.45F;
                    resetTimer = true;
                    break;
                }
            }
        }
    }
}

File: sources/Kevin Reborn/kevin/module/modules/player/Regen.kt
/*
 * This program is free software: you can redistribute it and/or modify
 *   it under the terms of the GNU General Public License as published by
 *   the Free Software Foundation, either version 3 of the License, or
 *   (at your option) any later version.
 *
 *   This program is distributed in the hope that it will be useful,
 *   but WITHOUT ANY WARRANTY; without even the implied warranty of
 *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *   GNU General Public License for more details.
 *
 *   You should have received a copy of the GNU General Public License
 *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package kevin.module.modules.player

import kevin.event.EventTarget
import kevin.event.UpdateEvent
import kevin.module.*
import kevin.utils.MovementUtils
import net.minecraft.network.play.client.C03PacketPlayer
import net.minecraft.potion.Potion
import java.util.*

class Regen : Module("Regen", "Regenerates your health much faster.", category = ModuleCategory.PLAYER) {

    private val modeValue = ListValue("Mode", arrayOf("Vanilla", "Spartan", "AAC4NoFire"), "Vanilla")
    private val healthValue = IntegerValue("Health", 18, 0, 20)
    private val foodValue = IntegerValue("Food", 18, 0, 20)
    private val speedValue = IntegerValue("Speed", 100, 1, 100)
    private val noAirValue = BooleanValue("NoAir", false)
    private val potionEffectValue = BooleanValue("PotionEffect", false)

    private var resetTimer = false

    @EventTarget
    fun onUpdate(event: UpdateEvent) {
        if (resetTimer)
            mc.timer.timerSpeed = 1F
        resetTimer = false

        val thePlayer = mc.thePlayer ?: return

        if ((!noAirValue.get() || thePlayer.onGround) && !thePlayer.capabilities.isCreativeMode &&
            thePlayer.foodStats.foodLevel > foodValue.get() && thePlayer.isEntityAlive && thePlayer.health < healthValue.get()) {
            if (potionEffectValue.get() && !thePlayer.isPotionActive(Potion.regeneration))
                return

            when (modeValue.get().lowercase(Locale.getDefault())) {
                "vanilla" -> {
                    repeat(speedValue.get()) {
                        mc.netHandler.addToSendQueue(C03PacketPlayer(thePlayer.onGround))
                    }
                }

                "spartan" -> {
                    if (MovementUtils.isMoving || !thePlayer.onGround)
                        return

                    repeat(9) {
                        mc.netHandler.addToSendQueue(C03PacketPlayer(thePlayer.onGround))
                    }

                    mc.timer.timerSpeed = 0.45F
                    resetTimer = true
                }

                "aac4nofire" -> {
                    if (mc.thePlayer!!.isBurning && mc.thePlayer!!.ticksExisted % 10 == 0) {
                        repeat(35) {
                            mc.netHandler.addToSendQueue(C03PacketPlayer(true))
                        }
                    }
                }
            }
        }
    }

    override val tag: String
        get() = modeValue.get()
}